pkg/governance-scripts/contracts/20230109-gauge-adder-migration-v2-to-v3/GaugeAdderMigrationCoordinator.sol

```diff
@@ -0,0 +1,163 @@
+        liquidityMiningCommitteeMultisig = _liquidityMiningCommitteeMultisig;
+        gaugeCheckpointingMultisig = _gaugeCheckpointingMultisig;
+
+        gaugeController = _newGaugeAdder.getGaugeController();
+        authorizerAdaptor = authorizerAdaptorEntrypoint.getAuthorizerAdaptor();
+    }
+
+    // Coordinator Setup
+
+    function _registerStages() internal override {
+        _registerStage(_firstStage);
+    }
+
+    function _firstStage() private {
+        _grantPermissionsOverBridgeParameters();
+        _setupNewGaugeAdder();
+        _deprecateOldGaugeAdder();
+    }
+
+    function _afterLastStage() internal virtual override {
+        ICurrentAuthorizer authorizer = ICurrentAuthorizer(address(getAuthorizer()));
+
+        authorizer.renounceRole(authorizer.DEFAULT_ADMIN_ROLE(), address(this));
+    }
+
+    // Internal functions
+
+    function _grantPermissionsOverBridgeParameters() private {
+        ICurrentAuthorizer authorizer = ICurrentAuthorizer(address(getAuthorizer()));
+
+        authorizer.grantRole(
+            IAuthentication(address(arbitrumRootGaugeFactory)).getActionId(
+                IArbitrumFeeProvider.setArbitrumFees.selector
+            ),
+            gaugeCheckpointingMultisig
+        );
+        authorizer.grantRole(
+            IAuthentication(address(optimismRootGaugeFactory)).getActionId(
+                IOptimismGasLimitProvider.setOptimismGasLimit.selector
+            ),
+            gaugeCheckpointingMultisig
+        );
+    }
+
+    function _setupNewGaugeAdder() private {
+        ICurrentAuthorizer authorizer = ICurrentAuthorizer(address(getAuthorizer()));
+
+        // Migrate factories from old GaugeAdder.
+        {
+            bytes32 addFactoryRole = IAuthentication(address(newGaugeAdder)).getActionId(
+                IGaugeAdder.addGaugeFactory.selector
+            );
+            authorizer.grantRole(addFactoryRole, address(this));
+
+            // Copy across gauge factories from previous GaugeAdder.
+            // Only one factory exists for each gauge type so this is sufficient.
+            newGaugeAdder.addGaugeFactory(
+                ILiquidityGaugeFactory(oldGaugeAdder.getFactoryForGaugeType(IGaugeAdder.GaugeType.Ethereum, 0)),
+                IGaugeAdder.GaugeType.Ethereum
+            );
+            newGaugeAdder.addGaugeFactory(
+                ILiquidityGaugeFactory(oldGaugeAdder.getFactoryForGaugeType(IGaugeAdder.GaugeType.Polygon, 0)),
+                IGaugeAdder.GaugeType.Polygon
+            );
+            newGaugeAdder.addGaugeFactory(
+                ILiquidityGaugeFactory(oldGaugeAdder.getFactoryForGaugeType(IGaugeAdder.GaugeType.Arbitrum, 0)),
+                IGaugeAdder.GaugeType.Arbitrum
+            );
+            newGaugeAdder.addGaugeFactory(
+                ILiquidityGaugeFactory(oldGaugeAdder.getFactoryForGaugeType(IGaugeAdder.GaugeType.Optimism, 0)),
+                IGaugeAdder.GaugeType.Optimism
+            );
```

Comment:
Are we sure that these are the capped gauge factories?

Reply:
These are definitely the old gauge factories but it also seems like they're still in use. Looking into this.

Reply:
Spoke to Solar and he's going to be using the capped gauge only from now on. We can just add the new gauges here.